homeassistant/components/prometheus/__init__.py

```diff
@@ -290,8 +293,11 @@ def state_as_number(state):
         try:
             value = state_helper.state_as_number(state)
         except ValueError:
-            _LOGGER.debug("Could not convert %s to float", state)
-            value = 0
+            if state.attributes.get(ATTR_DEVICE_CLASS) == SensorDeviceClass.TIMESTAMP:
```

Comment:
I wonder if this should be part of [state_helper.state_as_number](https://github.com/home-assistant/core/blob/128dadafaed5c49e989c599676743ee66ff01c80/homeassistant/helpers/state.py#L72) called above, making it useful for other monitoring integrations.

Reply:
Either move this code as suggested above into `state_helper.state_as_number` or into the `try` block before calling the `state_as_number` function.

The `ValueError` can be omitted by checking the device class first.

Reply:
Because moving it into `state_as_number` changes the nature of what `state_as_number` is in my mind (taking things like openings, presence, and other boolean-like states and transforming them into 1/0), I don't want to wrap those two changes into a single PR.

Moving the timestamp conversion into the try block is a good idea, coming up.

I think we still need to catch the ValueError (for both the timestamp and state_as_number), since the device class doesn't guarantee a valid state value.